apps/percona/functional/k8s_snapshot/test.yml

```diff
@@ -98,6 +114,30 @@
         - name: Creating snapshot.
           include_tasks: /funclib/kubectl/k8s_snapshot_clone/create_snapshot.yml
 
+        - block:
+
+            - name: Obtaining the storage class used from PVC
+              shell: kubectl get pvc {{ pvc_name }} -n {{ app_ns }} --no-headers -o custom-columns=:.spec.storageClassName
+              register: app_sc
+
+            - name: Obtaining the SPC from storage class
```

Comment:
to avoid 'grep' as much as possible, probably, you can look at cstor_delete_and_verify_pool_deployment.yml which uses one 'grep' to find the pool pod from pv. It first finds the deployments from CVRs, by selecting the CVR related to pv. And, it does a grep to find the pool pod from its deployment.
Actually, we can use 'replicaset' to even avoid this grep to find pool pod from its deployment.
```
kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name=="<pool_deployment>")].metadata.name}'
kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name=="<rs_name_from_above>")].metadata.name}'
```
In above, we are finding the replicaset which is controlled by that deployment, and, then, finding the pod which is controlled by that replicaset.

Reply:
probably, we may need to get the count also to check if there are multiple results (like one pod in terminating and another in containercreating etc) by using range like `-o=jsonpath='{range .items[*]}{.metadata.labels.cstorpool\.openebs\.io\/name}{"\n"}{end}'` and wc -l

Reply:
Modified kubectl commands to print the output in new lines (in case of multiple results as mentioned above) to use wc -l:
```
kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name=="<pool_deployment>")].metadata.name}{"\n"}'
kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name=="<rs_name_from_above>")].metadata.name}{"\n"}'
```
With this, we will be able to avoid grep